core-processor/src/ext.rs

```diff
@@ -384,36 +404,58 @@ impl EnvExt for Ext {
         self.return_and_store_err(result)
     }
 
-    fn reply_push(&mut self, buffer: &[u8]) -> Result<(), Self::Error> {
-        self.charge_gas_runtime(RuntimeCosts::ReplyPush(buffer.len() as u32))?;
-        let result = self.context.message_context.reply_push(buffer);
+    fn send_commit(
+        &mut self,
+        handle: u32,
+        msg: HandlePacket,
+        delay: u32,
+    ) -> Result<MessageId, Self::Error> {
+        self.charge_gas_runtime(RuntimeCosts::SendCommit(msg.payload().len() as u32))?;
+
+        self.check_forbidden_call(msg.destination())?;
+        self.charge_expiring_resources(&msg)?;
+
+        self.charge_sending_fee(delay)?;
+
+        let result = self
+            .context
+            .message_context
+            .send_commit(handle, msg, delay, None);
 
         self.return_and_store_err(result)
     }
 
-    fn send_commit(
+    fn reservation_send_commit(
```

Comment:
for all reservation sendings check that reservation exists, otherwise return err!